### DISCUSSION TITLE
[question] how to implement restoreState() for deep linking?

### DISCUSSION THREAD

**Discussion by chunghha** [csells/go_router#178] [Q&A]
Per @nyck33's question [here](https://github.com/csells/go_router/issues/142), I am trying to see if deep linking with my [example app](https://github.com/chunghha/fl_pages/commit/fc0a1d9789e08ba4fdcc5629cd206b536f317bb0) appearing to be working. However, I just realized that I am unsure about what to expect with oldBucket and how to restore my authState.

I did these steps.
- Initially '/' to '/login' redirected
- so on the login page, click the login button, goes to '/' 
- on the home page, enter '/polygon' to my browser's url to go to Polygons page.

I see deep linking was attempted by go_router but, because of the authStatus initialized, I ended up going to '/login' path.

@csells, could you review and advise how to use oldBucket for this use-case?

```bash
Performing hot restart...                                          252ms
Restarted application in 254ms.
|.. authState: AuthState(email: -, access_token: -, AuthStatus.unauthenticated)
|.. oldBucket null
\.... initialRestore true
GoRouter: known full paths for routes:
GoRouter:   => /
GoRouter:   => /polygon
GoRouter:   => /login
GoRouter: setting initial location /
|.. _routeToPath: /
\.... _authStatus: AuthStatus.unauthenticated
GoRouter: redirecting to /login
|.. _routeToPath: /login
\.... _authStatus: AuthStatus.unauthenticated
GoRouter: deep linking to /login
|.. _routeToPath: /login
\.... _authStatus: AuthStatus.unauthenticated
token 0002BIhGyCWXtb0YpIeRvED8TZ6R
GoRouter: going to /
|.. _routeToPath: /
\.... _authStatus: AuthStatus.authenticated
GoRouter: refreshing /
|.. _routeToPath: /
\.... _authStatus: AuthStatus.authenticated
|.. authState: AuthState(email: -, access_token: -, AuthStatus.unauthenticated)
|.. oldBucket null
\.... initialRestore true
GoRouter: known full paths for routes:
GoRouter:   => /
GoRouter:   => /polygon
GoRouter:   => /login
GoRouter: setting initial location /
|.. _routeToPath: /
\.... _authStatus: AuthStatus.unauthenticated
GoRouter: redirecting to /login
|.. _routeToPath: /login
\.... _authStatus: AuthStatus.unauthenticated
GoRouter: deep linking to /polygon
|.. _routeToPath: /polygon
\.... _authStatus: AuthStatus.unauthenticated
GoRouter: redirecting to /login
|.. _routeToPath: /login
\.... _authStatus: AuthStatus.unauthenticated
token 0002BIhGyCWXtb0YpIeRvED8TZ6R
```

**Comment by csells** [comment]
I'm not as familiar with state restoration as I should be. Perhaps [this sample](https://github.com/csells/go_router/blob/master/example/lib/state_restoration.dart) will be helpful?

**Reply by chunghha** [reply]
In the sample, the line 23 doesn't implement any so it's hardly to be a helpful example.

**Reply by csells** [reply]
Sorry about that.
